File: src/main/java/com/heybys/oddments/fooddelivery/service/order/OrderService.java
package com.heybys.oddments.fooddelivery.service.order;

import com.heybys.oddments.base.annotations.UseCase;
import com.heybys.oddments.fooddelivery.domain.cart.CartId;
import com.heybys.oddments.fooddelivery.domain.cart.CartRepository;
import com.heybys.oddments.fooddelivery.domain.order.OrderRepository;
import com.heybys.oddments.fooddelivery.domain.user.UserId;
import com.heybys.oddments.fooddelivery.domain.user.UserRepository;
import lombok.RequiredArgsConstructor;

@UseCase
@RequiredArgsConstructor
public class OrderService {

    private final UserRepository userRepository;
    private final CartRepository cartRepository;
    private final OrderRepository orderRepository;

    public void placeOrder(UserId userId, CartId cartId) {
        cartRepository.find(cartId);
    }
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/service/order/OrderService.kt
package com.heybys.oddments.fooddelivery.service.order

import com.heybys.oddments.base.annotations.UseCase
import com.heybys.oddments.fooddelivery.domain.cart.CartId
import com.heybys.oddments.fooddelivery.domain.cart.CartRepository
import com.heybys.oddments.fooddelivery.domain.order.OrderRepository
import com.heybys.oddments.fooddelivery.domain.user.UserId
import com.heybys.oddments.fooddelivery.domain.user.UserRepository

@UseCase
class OrderService(
    private val userRepository: UserRepository,
    private val cartRepository: CartRepository,
    private val orderRepository: OrderRepository,
) {

    fun placeOrder(userId: UserId, cartId: CartId) {
        cartRepository.find(cartId)
    }
}
